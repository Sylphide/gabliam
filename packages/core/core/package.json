{
  "name": "@gabliam/core",
  "version": "8.0.0-alpha.97+ec328fa",
  "description": "Gabliam is on top of inversify",
  "main": "lib/index.js",
  "typings": "lib/index.d.ts",
  "module": "lib/index.js",
  "scripts": {
    "build": "rimraf lib && tsc -p tsconfig.build.json"
  },
  "files": [
    "lib",
    "src"
  ],
  "dependencies": {
    "@gabliam/expression": "8.0.0-alpha.97+ec328fa",
    "@types/joi": "^14.0.1",
    "caller": "^1.0.1",
    "co": "^4.6.0",
    "debug": "^4.1.1",
    "gert": "^2.0.0",
    "gert-topo-sort": "^1.0.0",
    "glob": "^7.1.3",
    "inversify": "^5.0.1",
    "inversify-logger-middleware": "^3.0.0",
    "joi": "^14.3.1",
    "js-yaml": "^3.12.1",
    "lodash": "^4.17.11",
    "shortstop": "^1.0.3",
    "shortstop-handlers": "^1.0.1",
    "tslib": "^1.9.3"
  },
  "devDependencies": {
    "@types/debug": "^0.0.31",
    "@types/fs-extra": "^5.0.4",
    "@types/glob": "^7.1.1",
    "@types/js-yaml": "^3.11.4",
    "@types/lodash": "^4.14.119",
    "fs-extra": "^7.0.1",
    "properties": "^1.2.1",
    "rxjs": "^6.3.3",
    "toml": "^2.3.3"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/gabliam/gabliam.git"
  },
  "author": "David Touzet <dtouzet@gmail.com>",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/gabliam/gabliam/issues"
  },
  "homepage": "https://github.com/gabliam/gabliam/tree/master/packages/core#readme",
  "engines": {
    "node": ">=8.9.0"
  },
  "gitHead": "ec328fa0b62f474d470f74a2fc9637838f54091e"
}
