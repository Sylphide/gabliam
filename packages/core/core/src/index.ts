import 'reflect-metadata';
export {
  decorate,
  inject,
  injectable,
  interfaces as inversifyInterfaces,
  multiInject,
  named,
  optional,
  postConstruct,
  tagged,
  targetName,
  unmanaged,
} from 'inversify';
export {
  APP_CONFIG,
  CORE_CONFIG,
  INJECT_CONTAINER_KEY,
  VALUE_EXTRACTOR,
} from './constants';
export * from './container';
export * from './decorator';
export * from './errors';
export * from './gabliam';
export * from './interfaces';
export * from './joi';
export { configResolver, FileLoader, Resolver } from './loaders';
export * from './metadata';
export * from './promise-utils';
export * from './reflection';
export * from './registry';
export * from './utils';
