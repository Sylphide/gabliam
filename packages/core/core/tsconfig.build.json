{
  "extends": "../../../tsconfig.base",
  "compilerOptions": {
    "sourceMap": false,
    "outDir": "../../../dist/packages/core/core",
    "baseUrl": ".",
    "rootDir": ".",
    "paths": {
      "@gabliam/expression": ["../../../dist/packages/core/expression"]
    }
  },
  "include":[
    "index.ts",
    "src/**/*.ts"
  ]
}
