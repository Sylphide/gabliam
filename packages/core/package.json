{
  "name": "@gabliam/core",
  "version": "7.9.0",
  "description": "Gabliam is on top of inversify",
  "main": "lib/index.js",
  "typings": "lib/index.d.ts",
  "module": "lib/index.js",
  "scripts": {
    "build": "rimraf lib && tsc -p tsconfig.build.json"
  },
  "files": [
    "lib",
    "src"
  ],
  "dependencies": {
    "@gabliam/expression": "7.8.2",
    "@types/joi": "^13.0.8",
    "caller": "^1.0.1",
    "debug": "^3.0.0",
    "gert": "^2.0.0",
    "gert-topo-sort": "^1.0.0",
    "glob": "^7.1.1",
    "inversify": "^4.3.0",
    "inversify-logger-middleware": "^3.0.0",
    "joi": "^13.0.0",
    "js-yaml": "^3.8.1",
    "lodash": "^4.17.4",
    "shortstop": "^1.0.3",
    "shortstop-handlers": "^1.0.1",
    "tslib": "^1.8.0"
  },
  "devDependencies": {
    "@types/debug": "^0.0.30",
    "@types/glob": "^5.0.30",
    "@types/js-yaml": "^3.5.29",
    "@types/lodash": "^4.14.53"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/gabliam/gabliam.git"
  },
  "author": "David Touzet <dtouzet@gmail.com>",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/gabliam/gabliam/issues"
  },
  "homepage": "https://github.com/gabliam/gabliam/tree/master/packages/core#readme",
  "engines": {
    "node": ">=8.9.0"
  }
}
