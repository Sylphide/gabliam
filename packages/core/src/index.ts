import 'reflect-metadata';
export * from './interfaces';
export * from './decorators';
export * from './gabliam';
export {
  APP_CONFIG,
  CORE_CONFIG,
  VALUE_EXTRACTOR,
  INJECT_CONTAINER_KEY,
} from './constants';

export * from './registry';
export * from './errors';
export * from './utils';
export * from './joi';
export * from './container';
export { FileLoader, Resolver, configResolver } from './loaders';

export {
  injectable,
  tagged,
  named,
  inject,
  optional,
  unmanaged,
  multiInject,
  targetName,
  decorate,
  postConstruct,
  interfaces as inversifyInterfaces,
} from 'inversify';
