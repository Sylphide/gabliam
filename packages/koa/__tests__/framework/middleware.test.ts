import {
  Controller,
  Get,
  Post,
  Put,
  Patch,
  Head,
  Delete,
  All,
  koa,
  KoaConfig,
  Middleware,
  MiddlewareInject
} from '../../src/index';
import { KoaPluginTest } from '../koa-plugin-test';
import { Config } from '@gabliam/core';
import * as sinon from 'sinon';

let appTest: KoaPluginTest;

beforeEach(async () => {
  appTest = new KoaPluginTest();
});

afterEach(async () => {
  await appTest.destroy();
});

describe('Middleware:', () => {
  let result: string;
  const middleware: any = {
    a: async function(ctx: koa.Context, nextFunc: () => Promise<any>) {
      result += 'a';
      await nextFunc();
    },
    b: async function(ctx: koa.Context, nextFunc: () => Promise<any>) {
      result += 'b';
      await nextFunc();
    },
    c: async function(ctx: koa.Context, nextFunc: () => Promise<any>) {
      result += 'c';
      await nextFunc();
    }
  };
  const spyA = sinon.spy(middleware, 'a');
  const spyB = sinon.spy(middleware, 'b');
  const spyC = sinon.spy(middleware, 'c');

  beforeEach(() => {
    result = '';
    spyA.resetHistory();
    spyB.resetHistory();
    spyC.resetHistory();
  });

  test('should call method-level middleware correctly (GET)', async () => {
    @Controller('/')
    class TestController {
      @Get('/', spyA, spyB, spyC)
      public getTest(ctx: koa.Context) {
        ctx.body = 'GET';
      }
    }
    appTest.addClass(TestController);
    await appTest.build();

    const response = await appTest
      .supertest()
      .get('/')
      .expect(200);

    expect(spyA.calledOnce).toBe(true);
    expect(spyB.calledOnce).toBe(true);
    expect(spyC.calledOnce).toBe(true);
    expect(response).toMatchSnapshot();
    expect(result).toMatchSnapshot();
  });

  test('should call method-level middleware correctly (POST)', async () => {
    @Controller('/')
    class TestController {
      @Post('/', spyA, spyB, spyC)
      public postTest(ctx: koa.Context) {
        ctx.body = 'POST';
      }
    }

    appTest.addClass(TestController);
    await appTest.build();

    const response = await appTest
      .supertest()
      .post('/')
      .expect(200);
    expect(spyA.calledOnce).toBe(true);
    expect(spyB.calledOnce).toBe(true);
    expect(spyC.calledOnce).toBe(true);
    expect(response).toMatchSnapshot();
    expect(result).toMatchSnapshot();
  });

  test('should call method-level middleware correctly (PUT)', async () => {
    @Controller('/')
    class TestController {
      @Put('/', spyA, spyB, spyC)
      public postTest(ctx: koa.Context) {
        ctx.body = 'PUT';
      }
    }
    appTest.addClass(TestController);
    await appTest.build();

    const response = await appTest
      .supertest()
      .put('/')
      .expect(200);
    expect(spyA.calledOnce).toBe(true);
    expect(spyB.calledOnce).toBe(true);
    expect(spyC.calledOnce).toBe(true);
    expect(response).toMatchSnapshot();
    expect(result).toMatchSnapshot();
  });

  test('should call method-level middleware correctly (PATCH)', async () => {
    @Controller('/')
    class TestController {
      @Patch('/', spyA, spyB, spyC)
      public postTest(ctx: koa.Context) {
        ctx.body = 'PATCH';
      }
    }

    appTest.addClass(TestController);
    await appTest.build();

    const response = await appTest
      .supertest()
      .patch('/')
      .expect(200);
    expect(spyA.calledOnce).toBe(true);
    expect(spyB.calledOnce).toBe(true);
    expect(spyC.calledOnce).toBe(true);
    expect(response).toMatchSnapshot();
    expect(result).toMatchSnapshot();
  });

  test('should call method-level middleware correctly (HEAD)', async () => {
    @Controller('/')
    class TestController {
      @Head('/', spyA, spyB, spyC)
      public postTest(ctx: koa.Context) {
        ctx.body = 'HEAD';
      }
    }
    appTest.addClass(TestController);
    await appTest.build();

    const response = await appTest
      .supertest()
      .head('/')
      .expect(200);
    expect(spyA.calledOnce).toBe(true);
    expect(spyB.calledOnce).toBe(true);
    expect(spyC.calledOnce).toBe(true);
    expect(response).toMatchSnapshot();
    expect(result).toMatchSnapshot();
  });

  test('should call method-level middleware correctly (DELETE)', async () => {
    @Controller('/')
    class TestController {
      @Delete('/', spyA, spyB, spyC)
      public postTest(ctx: koa.Context) {
        ctx.body = 'DELETE';
      }
    }
    appTest.addClass(TestController);
    await appTest.build();

    const response = await appTest
      .supertest()
      .delete('/')
      .expect(200);
    expect(spyA.calledOnce).toBe(true);
    expect(spyB.calledOnce).toBe(true);
    expect(spyC.calledOnce).toBe(true);
    expect(response).toMatchSnapshot();
    expect(result).toMatchSnapshot();
  });

  test('should call method-level middleware correctly (ALL)', async () => {
    @Controller('/')
    class TestController {
      @All('/', spyA, spyB, spyC)
      public postTest(ctx: koa.Context) {
        ctx.body = 'ALL';
      }
    }
    appTest.addClass(TestController);
    await appTest.build();

    const response = await appTest
      .supertest()
      .get('/')
      .expect(200);
    expect(spyA.calledOnce).toBe(true);
    expect(spyB.calledOnce).toBe(true);
    expect(spyC.calledOnce).toBe(true);
    expect(response).toMatchSnapshot();
    expect(result).toMatchSnapshot();
  });

  test('should call controller-level middleware correctly', async () => {
    @Controller({
      path: '/',
      middlewares: [spyA, spyB, spyC]
    })
    class TestController {
      @Get('/')
      public getTest(ctx: koa.Context) {
        ctx.body = 'GET';
      }
    }

    appTest.addClass(TestController);
    await appTest.build();

    const response = await appTest
      .supertest()
      .get('/')
      .expect(200);
    expect(spyA.calledOnce).toBe(true);
    expect(spyB.calledOnce).toBe(true);
    expect(spyC.calledOnce).toBe(true);
    expect(response).toMatchSnapshot();
    expect(result).toMatchSnapshot();
  });

  test('should call server-level middleware correctly', async () => {
    @Controller('/')
    class TestController {
      @Get('/')
      public getTest(ctx: koa.Context) {
        ctx.body = 'GET';
      }
    }

    @Config()
    class ServerConfig {
      @KoaConfig()
      serverConfig(app: koa) {
        app.use(spyA);
        app.use(spyB);
        app.use(spyC);
      }
    }

    appTest.addClass(TestController);
    appTest.addClass(ServerConfig);
    await appTest.build();

    const response = await appTest
      .supertest()
      .get('/')
      .expect(200);
    expect(spyA.calledOnce).toBe(true);
    expect(spyB.calledOnce).toBe(true);
    expect(spyC.calledOnce).toBe(true);
    expect(response).toMatchSnapshot();
    expect(result).toMatchSnapshot();
  });

  test('should call all middleware in correct order', async () => {
    @Controller({
      path: '/',
      middlewares: [spyB]
    })
    class TestController {
      @Get('/', spyC)
      public getTest(ctx: koa.Context) {
        ctx.body = 'GET';
      }
    }

    @Config()
    class ServerConfig {
      @KoaConfig()
      serverConfig(app: koa) {
        app.use(spyA);
      }
    }

    appTest.addClass(TestController);
    appTest.addClass(ServerConfig);
    await appTest.build();

    const response = await appTest
      .supertest()
      .get('/')
      .expect(200);
    expect(spyA.calledOnce).toBe(true);
    expect(spyB.calledOnce).toBe(true);
    expect(spyC.calledOnce).toBe(true);
    expect(response).toMatchSnapshot();
    expect(result).toMatchSnapshot();
  });

  test('should resolve controller-level middleware', async () => {
    const symbolId = Symbol('spyA');
    const strId = 'spyB';

    @Controller({
      path: '/',
      middlewares: [symbolId, strId]
    })
    class TestController {
      @Get('/')
      public getTest(ctx: koa.Context) {
        ctx.body = 'GET';
      }
    }

    appTest.addClass(TestController);
    appTest.gab.container.bind<koa.Middleware>(symbolId).toConstantValue(spyA);
    appTest.gab.container.bind<koa.Middleware>(strId).toConstantValue(spyB);
    await appTest.build();

    const response = await appTest
      .supertest()
      .get('/')
      .expect(200);
    expect(spyA.calledOnce).toBe(true);
    expect(spyB.calledOnce).toBe(true);
    expect(response).toMatchSnapshot();
    expect(result).toMatchSnapshot();
  });

  test('should resolve method-level middleware', async () => {
    const symbolId = Symbol('spyA');
    const strId = 'spyB';

    @Controller('/')
    class TestController {
      @Get('/', symbolId, strId)
      public getTest(ctx: koa.Context) {
        ctx.body = 'GET';
      }
    }

    appTest.addClass(TestController);
    appTest.gab.container.bind<koa.Middleware>(symbolId).toConstantValue(spyA);
    appTest.gab.container.bind<koa.Middleware>(strId).toConstantValue(spyB);
    await appTest.build();

    const response = await appTest
      .supertest()
      .get('/')
      .expect(200);
    expect(spyA.calledOnce).toBe(true);
    expect(spyB.calledOnce).toBe(true);
    expect(response).toMatchSnapshot();
    expect(result).toMatchSnapshot();
  });

  test('should compose controller- and method-level middleware', async () => {
    const symbolId = Symbol('spyA');
    const strId = 'spyB';

    @Controller({
      path: '/',
      middlewares: [symbolId]
    })
    class TestController {
      @Get('/', strId)
      public getTest(ctx: koa.Context) {
        ctx.body = 'GET';
      }
    }

    appTest.addClass(TestController);
    appTest.gab.container.bind<koa.Middleware>(symbolId).toConstantValue(spyA);
    appTest.gab.container.bind<koa.Middleware>(strId).toConstantValue(spyB);
    await appTest.build();

    const response = await appTest
      .supertest()
      .get('/')
      .expect(200);
    expect(spyA.calledOnce).toBe(true);
    expect(spyB.calledOnce).toBe(true);
    expect(response).toMatchSnapshot();
    expect(result).toMatchSnapshot();
  });
});

describe('Middleware inject:', () => {
  let result: string;
  let args: string;
  @Config()
  class TestConfig {
    @Middleware('a')
    a() {
      return () => async (ctx: koa.Context, nextFunc: () => Promise<any>) => {
        result += 'a';
        await nextFunc();
      };
    }

    @Middleware('b')
    b() {
      return () => async (ctx: koa.Context, nextFunc: () => Promise<any>) => {
        result += 'b';
        await nextFunc();
      };
    }

    @Middleware('c')
    c(arg?: string) {
      return () => async (ctx: koa.Context, nextFunc: () => Promise<any>) => {
        result += 'c';
        if (arg) {
          args += arg;
        }
        await nextFunc();
      };
    }

    @Middleware('de')
    de() {
      return (arg?: string, arg2?: string) => [
        async (ctx: koa.Context, nextFunc: () => Promise<any>) => {
          result += 'd';
          if (arg) {
            args += 'd' + arg;
          }
          if (arg2) {
            args += 'd' + arg2;
          }
          await nextFunc();
        },
        async (ctx: koa.Context, nextFunc: () => Promise<any>) => {
          result += 'e';
          if (arg) {
            args += 'e' + arg;
          }
          if (arg2) {
            args += 'e' + arg2;
          }
          await nextFunc();
        }
      ];
    }
  }

  beforeEach(() => {
    result = '';
    args = '';
  });

  test('should call method-level middleware correctly (GET)', async () => {
    @Controller('/')
    class TestController {
      @MiddlewareInject('de', 'dearg', 'dearg2')
      @MiddlewareInject('c', 'carg')
      @MiddlewareInject('b')
      @MiddlewareInject('a')
      @Get('/')
      public getTest(ctx: koa.Context) {
        ctx.body = 'GET';
      }
    }

    appTest.addClass(TestConfig);
    appTest.addClass(TestController);
    await appTest.build();

    const response = await appTest
      .supertest()
      .get('/')
      .expect(200);
    expect(response).toMatchSnapshot();
    expect(result).toMatchSnapshot();
    expect(args).toMatchSnapshot();
  });

  test('should call method-level middleware correctly (POST)', async () => {
    @Controller('/')
    class TestController {
      @MiddlewareInject('de', 'dearg', 'dearg2')
      @MiddlewareInject('c', 'carg')
      @MiddlewareInject('b')
      @MiddlewareInject('a')
      @Post('/')
      public postTest(ctx: koa.Context) {
        ctx.body = 'POST';
      }
    }

    appTest.addClass(TestConfig);
    appTest.addClass(TestController);
    await appTest.build();

    const response = await appTest
      .supertest()
      .post('/')
      .expect(200);
    expect(response).toMatchSnapshot();
    expect(result).toMatchSnapshot();
    expect(args).toMatchSnapshot();
  });

  test('should call method-level middleware correctly (PUT)', async () => {
    @Controller('/')
    class TestController {
      @MiddlewareInject('de', 'dearg', 'dearg2')
      @MiddlewareInject('c', 'carg')
      @MiddlewareInject('b')
      @MiddlewareInject('a')
      @Put('/')
      public postTest(ctx: koa.Context) {
        ctx.body = 'PUT';
      }
    }

    appTest.addClass(TestConfig);
    appTest.addClass(TestController);
    await appTest.build();

    const response = await appTest
      .supertest()
      .put('/')
      .expect(200);
    expect(response).toMatchSnapshot();
    expect(result).toMatchSnapshot();
    expect(args).toMatchSnapshot();
  });

  test('should call method-level middleware correctly (PATCH)', async () => {
    @Controller('/')
    class TestController {
      @MiddlewareInject('de', 'dearg', 'dearg2')
      @MiddlewareInject('c', 'carg')
      @MiddlewareInject('b')
      @MiddlewareInject('a')
      @Patch('/')
      public postTest(ctx: koa.Context) {
        ctx.body = 'PATCH';
      }
    }

    appTest.addClass(TestConfig);
    appTest.addClass(TestController);
    await appTest.build();

    const response = await appTest
      .supertest()
      .patch('/')
      .expect(200);
    expect(response).toMatchSnapshot();
    expect(result).toMatchSnapshot();
    expect(args).toMatchSnapshot();
  });

  test('should call method-level middleware correctly (HEAD)', async () => {
    @Controller('/')
    class TestController {
      @MiddlewareInject('de', 'dearg', 'dearg2')
      @MiddlewareInject('c', 'carg')
      @MiddlewareInject('b')
      @MiddlewareInject('a')
      @Head('/')
      public postTest(ctx: koa.Context) {
        ctx.body = 'HEAD';
      }
    }
    appTest.addClass(TestConfig);
    appTest.addClass(TestController);
    await appTest.build();

    const response = await appTest
      .supertest()
      .head('/')
      .expect(200);
    expect(response).toMatchSnapshot();
    expect(result).toMatchSnapshot();
    expect(args).toMatchSnapshot();
  });

  test('should call method-level middleware correctly (DELETE)', async () => {
    @Controller('/')
    class TestController {
      @MiddlewareInject('de', 'dearg', 'dearg2')
      @MiddlewareInject('c', 'carg')
      @MiddlewareInject('b')
      @MiddlewareInject('a')
      @Delete('/')
      public postTest(ctx: koa.Context) {
        ctx.body = 'DELETE';
      }
    }

    appTest.addClass(TestConfig);
    appTest.addClass(TestController);
    await appTest.build();

    const response = await appTest
      .supertest()
      .delete('/')
      .expect(200);
    expect(response).toMatchSnapshot();
    expect(result).toMatchSnapshot();
    expect(args).toMatchSnapshot();
  });

  test('should call method-level middleware correctly (ALL)', async () => {
    @Controller('/')
    class TestController {
      @MiddlewareInject('de', 'dearg', 'dearg2')
      @MiddlewareInject('c', 'carg')
      @MiddlewareInject('b')
      @MiddlewareInject('a')
      @All('/')
      public postTest(ctx: koa.Context) {
        ctx.body = 'ALL';
      }
    }

    appTest.addClass(TestConfig);
    appTest.addClass(TestController);
    await appTest.build();

    const response = await appTest
      .supertest()
      .get('/')
      .expect(200);
    expect(response).toMatchSnapshot();
    expect(result).toMatchSnapshot();
    expect(args).toMatchSnapshot();
  });

  test('should call controller-level middleware correctly', async () => {
    @MiddlewareInject('de', 'dearg', 'dearg2')
    @MiddlewareInject('c', 'carg')
    @MiddlewareInject('b')
    @MiddlewareInject('a')
    @Controller({
      path: '/'
    })
    class TestController {
      @Get('/')
      public getTest(ctx: koa.Context) {
        ctx.body = 'GET';
      }
    }

    appTest.addClass(TestConfig);
    appTest.addClass(TestController);
    await appTest.build();

    const response = await appTest
      .supertest()
      .get('/')
      .expect(200);
    expect(response).toMatchSnapshot();
    expect(result).toMatchSnapshot();
    expect(args).toMatchSnapshot();
  });

  test('should call all middleware in correct order', async () => {
    @MiddlewareInject('b')
    @MiddlewareInject('a')
    @Controller({
      path: '/'
    })
    class TestController {
      @MiddlewareInject('de', 'dearg', 'dearg2')
      @MiddlewareInject('c', 'carg')
      @Get('/')
      public getTest(ctx: koa.Context) {
        ctx.body = 'GET';
      }
    }

    appTest.addClass(TestController);
    appTest.addClass(TestConfig);
    await appTest.build();

    const response = await appTest
      .supertest()
      .get('/')
      .expect(200);
    expect(response).toMatchSnapshot();
    expect(result).toMatchSnapshot();
    expect(args).toMatchSnapshot();
  });
});
