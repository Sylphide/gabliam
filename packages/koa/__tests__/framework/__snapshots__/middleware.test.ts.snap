// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Middleware inject: should call all middleware in correct order 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "3",
    "content-type": "text/plain; charset=utf-8",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {},
    "method": "GET",
  },
  "status": 200,
  "text": "GET",
}
`;

exports[`Middleware inject: should call all middleware in correct order 2`] = `"abcde"`;

exports[`Middleware inject: should call all middleware in correct order 3`] = `"ddeargddearg2edeargedearg2"`;

exports[`Middleware inject: should call controller-level middleware correctly 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "3",
    "content-type": "text/plain; charset=utf-8",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {},
    "method": "GET",
  },
  "status": 200,
  "text": "GET",
}
`;

exports[`Middleware inject: should call controller-level middleware correctly 2`] = `"abcde"`;

exports[`Middleware inject: should call controller-level middleware correctly 3`] = `"ddeargddearg2edeargedearg2"`;

exports[`Middleware inject: should call method-level middleware correctly (ALL) 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "3",
    "content-type": "text/plain; charset=utf-8",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {},
    "method": "GET",
  },
  "status": 200,
  "text": "ALL",
}
`;

exports[`Middleware inject: should call method-level middleware correctly (ALL) 2`] = `"abcde"`;

exports[`Middleware inject: should call method-level middleware correctly (ALL) 3`] = `"ddeargddearg2edeargedearg2"`;

exports[`Middleware inject: should call method-level middleware correctly (DELETE) 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "6",
    "content-type": "text/plain; charset=utf-8",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {},
    "method": "DELETE",
  },
  "status": 200,
  "text": "DELETE",
}
`;

exports[`Middleware inject: should call method-level middleware correctly (DELETE) 2`] = `"abcde"`;

exports[`Middleware inject: should call method-level middleware correctly (DELETE) 3`] = `"ddeargddearg2edeargedearg2"`;

exports[`Middleware inject: should call method-level middleware correctly (GET) 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "3",
    "content-type": "text/plain; charset=utf-8",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {},
    "method": "GET",
  },
  "status": 200,
  "text": "GET",
}
`;

exports[`Middleware inject: should call method-level middleware correctly (GET) 2`] = `"abcde"`;

exports[`Middleware inject: should call method-level middleware correctly (GET) 3`] = `"ddeargddearg2edeargedearg2"`;

exports[`Middleware inject: should call method-level middleware correctly (HEAD) 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "4",
    "content-type": "text/plain; charset=utf-8",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {},
    "method": "HEAD",
  },
  "status": 200,
  "text": undefined,
}
`;

exports[`Middleware inject: should call method-level middleware correctly (HEAD) 2`] = `"abcde"`;

exports[`Middleware inject: should call method-level middleware correctly (HEAD) 3`] = `"ddeargddearg2edeargedearg2"`;

exports[`Middleware inject: should call method-level middleware correctly (PATCH) 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "5",
    "content-type": "text/plain; charset=utf-8",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {},
    "method": "PATCH",
  },
  "status": 200,
  "text": "PATCH",
}
`;

exports[`Middleware inject: should call method-level middleware correctly (PATCH) 2`] = `"abcde"`;

exports[`Middleware inject: should call method-level middleware correctly (PATCH) 3`] = `"ddeargddearg2edeargedearg2"`;

exports[`Middleware inject: should call method-level middleware correctly (POST) 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "4",
    "content-type": "text/plain; charset=utf-8",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {},
    "method": "POST",
  },
  "status": 200,
  "text": "POST",
}
`;

exports[`Middleware inject: should call method-level middleware correctly (POST) 2`] = `"abcde"`;

exports[`Middleware inject: should call method-level middleware correctly (POST) 3`] = `"ddeargddearg2edeargedearg2"`;

exports[`Middleware inject: should call method-level middleware correctly (PUT) 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "3",
    "content-type": "text/plain; charset=utf-8",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {},
    "method": "PUT",
  },
  "status": 200,
  "text": "PUT",
}
`;

exports[`Middleware inject: should call method-level middleware correctly (PUT) 2`] = `"abcde"`;

exports[`Middleware inject: should call method-level middleware correctly (PUT) 3`] = `"ddeargddearg2edeargedearg2"`;

exports[`Middleware: should call all middleware in correct order 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "3",
    "content-type": "text/plain; charset=utf-8",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {},
    "method": "GET",
  },
  "status": 200,
  "text": "GET",
}
`;

exports[`Middleware: should call all middleware in correct order 2`] = `"abc"`;

exports[`Middleware: should call controller-level middleware correctly 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "3",
    "content-type": "text/plain; charset=utf-8",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {},
    "method": "GET",
  },
  "status": 200,
  "text": "GET",
}
`;

exports[`Middleware: should call controller-level middleware correctly 2`] = `"abc"`;

exports[`Middleware: should call method-level middleware correctly (ALL) 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "3",
    "content-type": "text/plain; charset=utf-8",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {},
    "method": "GET",
  },
  "status": 200,
  "text": "ALL",
}
`;

exports[`Middleware: should call method-level middleware correctly (ALL) 2`] = `"abc"`;

exports[`Middleware: should call method-level middleware correctly (DELETE) 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "6",
    "content-type": "text/plain; charset=utf-8",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {},
    "method": "DELETE",
  },
  "status": 200,
  "text": "DELETE",
}
`;

exports[`Middleware: should call method-level middleware correctly (DELETE) 2`] = `"abc"`;

exports[`Middleware: should call method-level middleware correctly (GET) 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "3",
    "content-type": "text/plain; charset=utf-8",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {},
    "method": "GET",
  },
  "status": 200,
  "text": "GET",
}
`;

exports[`Middleware: should call method-level middleware correctly (GET) 2`] = `"abc"`;

exports[`Middleware: should call method-level middleware correctly (HEAD) 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "4",
    "content-type": "text/plain; charset=utf-8",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {},
    "method": "HEAD",
  },
  "status": 200,
  "text": undefined,
}
`;

exports[`Middleware: should call method-level middleware correctly (HEAD) 2`] = `"abc"`;

exports[`Middleware: should call method-level middleware correctly (PATCH) 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "5",
    "content-type": "text/plain; charset=utf-8",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {},
    "method": "PATCH",
  },
  "status": 200,
  "text": "PATCH",
}
`;

exports[`Middleware: should call method-level middleware correctly (PATCH) 2`] = `"abc"`;

exports[`Middleware: should call method-level middleware correctly (POST) 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "4",
    "content-type": "text/plain; charset=utf-8",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {},
    "method": "POST",
  },
  "status": 200,
  "text": "POST",
}
`;

exports[`Middleware: should call method-level middleware correctly (POST) 2`] = `"abc"`;

exports[`Middleware: should call method-level middleware correctly (PUT) 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "3",
    "content-type": "text/plain; charset=utf-8",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {},
    "method": "PUT",
  },
  "status": 200,
  "text": "PUT",
}
`;

exports[`Middleware: should call method-level middleware correctly (PUT) 2`] = `"abc"`;

exports[`Middleware: should call server-level middleware correctly 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "3",
    "content-type": "text/plain; charset=utf-8",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {},
    "method": "GET",
  },
  "status": 200,
  "text": "GET",
}
`;

exports[`Middleware: should call server-level middleware correctly 2`] = `"abc"`;

exports[`Middleware: should compose controller- and method-level middleware 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "3",
    "content-type": "text/plain; charset=utf-8",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {},
    "method": "GET",
  },
  "status": 200,
  "text": "GET",
}
`;

exports[`Middleware: should compose controller- and method-level middleware 2`] = `"ab"`;

exports[`Middleware: should resolve controller-level middleware 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "3",
    "content-type": "text/plain; charset=utf-8",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {},
    "method": "GET",
  },
  "status": 200,
  "text": "GET",
}
`;

exports[`Middleware: should resolve controller-level middleware 2`] = `"ab"`;

exports[`Middleware: should resolve method-level middleware 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "3",
    "content-type": "text/plain; charset=utf-8",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {},
    "method": "GET",
  },
  "status": 200,
  "text": "GET",
}
`;

exports[`Middleware: should resolve method-level middleware 2`] = `"ab"`;
